packages/core/src/State.ts

```diff
@@ -289,6 +312,32 @@ export class State<
    * The next events that will cause a transition from the current state.
    */
   public get nextEvents(): Array<TEvent['type']> {
-    return memo(this, 'nextEvents', () => nextEvents(this.configuration));
+    return memo(this, 'nextEvents', () => {
+      return [
+        ...new Set(flatten([...this.configuration.map((sn) => sn.ownEvents)]))
+      ];
+    });
+  }
+
+  public get meta() {
+    return this.configuration.reduce((acc, stateNode) => {
+      if (stateNode.meta !== undefined) {
+        acc[stateNode.id] = stateNode.meta;
+      }
+      return acc;
+    }, {} as Record<string, any>);
+  }
+
+  public get tags(): Set<string> {
+    return new Set(flatten(this.configuration.map((sn) => sn.tags)));
+  }
+
+  public clone(
```

Comment:
perhaps a better name for this would be `assign`? I also feel that this should be hidden as an internal util or something, public users shouldn't use this

Reply:
👋

Reply:
Not `assign` for obvious reasons, but I'll figure out a different name. Even if it's `clone`, it will be internal.

Reply:
it's never internal if it's a public method on `State` - that's something that we directly hand to the user

Reply:
dd5cadc4ec44cc80b5c6c3205fcc950ff8a67bad